data/publiccloud/azure_vn.sh

```diff
@@ -0,0 +1,420 @@
+echo "Created container $STORAGE_CONT"
+
+count=$(( $number_of_nics - 1 ))
+while [ $count -ge 0 ]
+do
+    subnet_names[$count]=$base_subnet"$count"
+    nic_names[$count]=$base_nic"$count"
+    count=$(( $count - 1 ))
+done
+
+skip=1
+echo "Check vnet exists for $grpname and $vnet { az network vnet show }"
+az network vnet show -g $grpname  --name  $vnet > /dev/null 2>&1 && skip=2;echo "Vnet \"$vnet\" already exists, skipping vnet and subnet creation"|| skip=1
+cmd_status "az_network_vnet_show" az network vnet show -g "$grpname"  --name "$vnet"
+echo "Skip value $skip"
+if [ $skip -eq 1 ]; then
+    # Create VNET
+    echo "Creating Azure virtual network $vnet { az network vnet create } "
+    az network vnet create \
+        --resource-group $grpname \
+        --name $vnet \
+        --address-prefix $addressprefix \
+        --location $location \
+        --output table
+
+    echo "Done creating Azure virtual network $vnet"
+
+    # Create as many subnets as there are NICs
+    i=0
+    echo "subnetprefix" ${subnet_prefixes[@]}
+    for prefix in "${subnet_prefixes[@]}"
+    do
+        echo "Creating virtual subnet ${subnet_names[$i]} $prefix..{az network vnet subnet create }"
+        cmd_status "az_network_vnet_subnet_create" az network vnet subnet create \
+            --address-prefix "$prefix" \
+            --name "${subnet_names[$i]}" \
+            --resource-group "$grpname" \
+            --vnet-name "$vnet" \
+            --output table
+        echo "Done creating subnet ${subnet_names[$i]} with prefix $prefix and $i"
+	let "i+=1"
+    done
+fi
+
+#
+# Creating routing tables for vMX WAN ports (add tables as needed for more than 2 wan ports)
+#
+echo "Creating routing tables.{ az network route-table create }"
+cmd_status "az_network_route_table_create1" az network route-table create \
+    --name "$grpname-rt-to-subnet2" \
+    --resource-group "$grpname" \
+    --location "$location" \
+    --output table
+cmd_status "az_network_route_table_create2" az network route-table create \
+    --name "$grpname-rt-to-subnet3" \
+    --resource-group "$grpname" \
+    --location "$location" \
+    --output table
+
+echo "Subnet update.{ az network vnet subnet update }"
+cmd_status "az_network_vnet_subnet_update1" az network vnet subnet update \
+    --resource-group "$grpname" \
+    --vnet-name "$vnet" \
+    --name "$vnet-vsubnet0" \
+    --route-table "$grpname-rt-to-subnet3" \
+    --output table
+cmd_status "az_network_vnet_subnet_update2" az network vnet subnet update \
+     --resource-group "$grpname" \
+     --vnet-name "$vnet" \
+     --name "$vnet-vsubnet1" \
+     --route-table "$grpname-rt-to-subnet2" \
+     --output table
+
+#
+# Create all NICs
+#
+
+echo "Creating public IP addresses and NICs.{ az network public-ip create } { az network nic create }"
+
+i=0
+allnics=""
+for nic in "${nic_names[@]}"
```

Comment:
basically you have a loop where in each iteration you executing totally different piece of code => which means you don't need this loop just create 3 nic's one after another

Reply:
ok